client/src/app/components/clt-editor/shared/clt-editor.service.ts

```diff
@@ -0,0 +1,16 @@
+import { Injectable } from '@angular/core';
+import { Subject }    from 'rxjs/Subject';
+import {SidebarEvent} from "../../sidebar/shared/sidebar.events";
+
+@Injectable()
+export class CltEditorService {
+    // Observable sources
+    private sidebarEvents = new Subject<SidebarEvent>();
+    
+    // Observable streams
+    public publishedSidebarEvents = this.sidebarEvents.asObservable();
```

Comment:
No need for two observables of the same data.
Superfluous naming (all events are published, there can be no unpublishedSidebarEvents).

Don't declare attributes that take values from other attributes, it makes code work or not depending on the ordering of the attributes.

For example  
```typescript
private sidebarEvents = new Subject<SidebarEvent>();
public publishedSidebarEvents = this.sidebarEvents.asObservable();
```
will work, while  
```typescript
public publishedSidebarEvents = this.sidebarEvents.asObservable();
private sidebarEvents = new Subject<SidebarEvent>();
```
will not.